webapp/src/utils/jira_issue_metadata.tsx

```diff
@@ -376,23 +376,25 @@ export function generateJQLStringFromSubscriptionFilters(issueMetadata: IssueMet
     return [projectJQL, issueTypesJQL, filterFieldsJQL].filter(Boolean).join(' AND ');
 }
 
-export function jiraIssueToReducer(data: TicketData): TicketDetails | null {
+export function getJiraTicketDetails(data?: TicketData): TicketDetails | null {
     if (!data) {
         return null;
     }
 
     const assignee = data && data.fields && data.fields.assignee ? data.fields.assignee : null;
     const ticketDetails: TicketDetails = {
+
+        // TODO: Add optional chaining operator
         assigneeName: (assignee && assignee.displayName) || '',
         assigneeAvatar: (assignee && assignee.avatarUrls && assignee.avatarUrls['48x48']) || '',
-        labels: data.fields.labels,
-        description: data.fields.description,
-        summary: data.fields.summary,
+        labels: data.fields && data.fields.labels,
+        description: data.fields && data.fields.description,
+        summary: data.fields && data.fields.summary,
         ticketId: data.key,
-        jiraIcon: data.fields.project.avatarUrls && data.fields.project.avatarUrls['48x48'],
-        versions: data.fields.versions.length ? data.fields.versions[0] : '',
-        statusKey: data.fields.status.name,
-        issueIcon: data.fields.issuetype.iconUrl,
+        jiraIcon: data.fields && data.fields.project && data.fields.project.avatarUrls && data.fields.project.avatarUrls['48x48'],
+        versions: data.fields && data.fields.versions && data.fields.versions.length ? data.fields.versions[0] : '',
+        statusKey: data.fields && data.fields.status && data.fields.status.name,
+        issueIcon: data.fields && data.fields.issuetype && data.fields.issuetype.iconUrl,
```

Comment:
Since values in `TicketDataFields` can be `undefined` (e.g. the `versions` field), I think we should update the `TicketDataFields` type to reflect that, and then typescript will enforce that we return all defined things here.

Reply:
Values like `jiraIcon` may throw an error when giving an `img` element an undefined `src` prop, so we want to make sure these are all strings. Also, we probably want to have fallback images in case those are not defined.